common/src/main/java/org/alfresco/hxi_connector/common/model/repository/Node.java

```diff
@@ -0,0 +1,41 @@
+/*-
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 - 2024 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.common.model.repository;
+
+import java.util.List;
+import jakarta.validation.constraints.NotBlank;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public record Node(@NotBlank String id, @JsonProperty("aspectNames") List<String> aspects)
```

Comment:
I expected this information to be inside an object called `Prediction` or maybe `NodePrediction`.

Reply:
This should represent the object which is passed to `/nodes` API - if you think this is not clear enough I don't mind to rename it - `NodePrediction` should be OK IMO

Reply:
I think `Node` is okay here as it is the node that gets send to ACS- but that's just my opinion.

Reply:
I didn't realise this was the object that was being passed to the `/nodes` API. Please leave it as is. I think it will become more obvious as we add support for populating more fields.